### DISCUSSION TITLE
What is the format of Prisma file (.prisma)?

### DISCUSSION THREAD

**Discussion by tal-rofe** [prisma/prisma#19113] [Q&A] [answered]
Just for my knowledge, is it some known format? at first glance it is similar to gRPC. Does it extend some known one?
Does Prisma has it own format?

**Comment by nurul3101** [comment]
Hello @tal-rofe 👋 

Prisma has its own format called the Prisma Schema Language (PSL) for defining database models and configurations. The Prisma schema is a `.prisma` file that uses the PSL to define your application's data model and other configurations.

It is specifically designed for Prisma and serves as a single source of truth for both your database schema and the generated Prisma Client. The PSL syntax is unique to Prisma and doesn't directly extend any other known formats. 

Other community-built Prisma clients like

- Python: https://prisma-client-py.readthedocs.io/en/stable/ 
- Rust: https://github.com/Brendonovich/prisma-client-rust 

also leverage this schema file as a single source of truth.

**Reply by tal-rofe** [reply]
@nurul3101 
Thanks for response. Could you please tell where is the source code (which is open-source) of the schema parsing (the code that parses the schema file)

**Reply by nurul3101** [reply]
You are probably interested in this repository:
- https://github.com/prisma/language-tools

**Reply by tal-rofe** [reply]
@nurul3101 
Maybe I'm mistaken, but I'm not sure this the thing I'm looking for.
I'm looking for the source code that parses the `schema.prisma` file,

maybe this one?:
https://github.com/prisma/prisma-engines/tree/main/prisma-fmt/src

**Reply by nurul3101** [reply, marked as answer]
The language tools repository is for our Extensions like for VS Code.
For parsing logic, prisma-engines would be the correct place.
